dd-sdk-android/src/test/kotlin/com/datadog/android/utils/forge/SpanForgeryFactory.kt

```diff
@@ -0,0 +1,57 @@
+package com.datadog.android.utils.forge
+
+import com.nhaarman.mockitokotlin2.mock
+import datadog.opentracing.DDSpan
+import datadog.opentracing.DDTracer
+import datadog.trace.api.Config
+import fr.xgouchet.elmyr.Forge
+import fr.xgouchet.elmyr.ForgeryFactory
+import io.opentracing.Tracer
+
+internal class SpanForgeryFactory : ForgeryFactory<DDSpan> {
+
+    companion object {
+        val TEST_TRACER = DDTracer(Config.get(), mock())
+    }
+
+    override fun getForgery(forge: Forge): DDSpan {
+        val operationName = forge.anAlphabeticalString()
+        val resourceName = forge.anAlphabeticalString()
+        val serviceName = forge.anAlphabeticalString()
+        val spanType = forge.anAlphabeticalString()
+        val isWithErrorFlag = forge.aBool()
+        val tags = forge.exhaustiveTraceTags()
+        val spanBuilder = TEST_TRACER
+            .buildSpan(operationName)
+            .withSpanType(spanType)
+            .withResourceName(resourceName)
+            .withServiceName(serviceName)
+
+        if (isWithErrorFlag) {
+            spanBuilder.withErrorFlag()
+        }
+        (spanBuilder as Tracer.SpanBuilder).apply {
```

Comment:
why do you need this cast ? If you remove it, you can also remove the `apply` which makes this part less readable

Reply:
Beats me why I did that honestly, I know I was having problems calling the right method there and wanted to make sure I call the one from the interface, but yeah I can remove that of course.